dotcom-rendering/src/web/components/LightboxHash.importable.tsx

```diff
@@ -0,0 +1,27 @@
+import { log } from '@guardian/libs';
+import { useEffect } from 'react';
+
+/**
```

Comment:
It'd be nice if we could have a 'hasHash' option for loading islands where we don't necessarily care what the hash is, but at least only have to download & execute the code if there is actually a hash

Reply:
I was talking to @arelra about this. I feel like these sorts of javascript snippets would ideally not live in islands and there is probably a better, more efficient, way to execute this code on the client. 

Maybe one island with common js blobs in it. This is basically excatly [what Frontend does](https://github.com/guardian/frontend/blob/f89072cb8ac3fcc970870acb747f292e3b0584da/static/src/javascripts/bootstraps/enhanced/common.js). But you could also use an Islets pattern like this [PR proposed](https://github.com/guardian/gateway/pull/1513).

Or, you know, a script tag 🤷

Reply:
So I've implemented the hydrate on hash pattern (removes the need for multiple islands, keeps the url as the source of truth) but I've only done it in a naive way. Basically like this `window.location.hash.includes(name) || hasLightboxHash(name)`. This works, and might be sufficient for now, but a better api could be developed to make it easier to do similar things in future.

Maybe 

```
deferUntil='hash'
hashPattern="some kind of matcher" // ?
```

But maybe the lightbox use case is unique and an OR statement is all you need?